web/app/js/components/Navigation.jsx

```diff
@@ -166,12 +174,42 @@ class NavigationBase extends React.Component {
       }).catch(this.handleApiError);
   }
 
+  fetchLatestCommunityUpdate() {
+    let jsonFeedUrl = "https://linkerd.io/dashboard/index.json";
+    this.communityUpdatesPromise = fetch(jsonFeedUrl)
+      .then(rsp => rsp.json())
+      .then(rsp => rsp.data.date)
+      .then(rsp => {
+        if (rsp.length > 0) {
+          let lastClicked = localStorage["linkerd-updates-last-checked"];
+          if (!lastClicked) {
+            localStorage.setItem("linkerd-updates-last-checked", new Date());
+            this.setState({ hideUpdateBadge: false });
+          } else {
+            let latestCommunityUpdate = rsp.map(update => new Date(update.date))
+              .sort((a, b) => b - a)[0];
+            if (latestCommunityUpdate > lastClicked) {
```

Comment:
is this doing a string comparison? would this be easier as an integer timestamp?
```js
new Date().getTime()
1552084942276
```

Reply:
On lines 189-190, I'm creating a map of `Date` instances from the returned JSON dates, then passing a `compareFunction` into `sort()` that subtracts them to get a value that is either negative, positive or zero. 

This is so I can sort in descending order (latest-earliest) which requires me to use a `compareFunction`.

Using `getTime()` on the `Date` instances would work too, but since it would add an extra step and I would still need a `compareFunction` I'm inclined to keep as-is.

Reply:
Hmm, I was mostly curious about `lastClicked`, which `localStorage["linkerd-updates-last-clicked"]`, which I think is a string? I'm not totally sure what happens in `latestCommunityUpdate > lastClicked` if its comparing a `Date` with a string?

Reply:
You are totally right, thanks for clarifying! Let me fix now.

Reply:
Confirmed this all works, thanks for the fix.

For a bit more context, I'm generally paranoid about storing date strings and opt for integer timestamps in UTC. JS's `Date.getTime()` is milliseconds since the unix epoch, not quite the same as unix timestamps (seconds since epoch), but close. It eliminates format-specific parsing or consideration for timezones, and makes sorting/comparing easier, particularly across languages.

Going forward we'll want to be mindful that we're syndicating https://linkerd.io/dashboard/index.json timestamps as [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601), and storing [ECMA-262](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toString#Description) in `localstorage`. Again this is all pretty academic, just want to be intentional about supporting this.